### PR TITLE
OtelFiber: Add Next config setting

### PR DESCRIPTION
**Pull request by emanuelef**
Just an idea for https://github.com/gofiber/contrib/issues/632
If this makes sense I can go on with this idea and start testing it.

### CODE REVIEW

**Comment by ReneWerner87** [comment]
Tests and readme adjustments are missing

**Comment by ReneWerner87** [comment]
Why not using the skip middleware or a next function like in all other middlewares

**Comment by emanuelef** [comment]
Thanks for looking into that.
This approach is ok for what I need but let me know if there's a better way.
I looked at otelzap and did something similar.

**Comment by emanuelef** [comment]
@ReneWerner87 just wanted to ask if this feature is relevant, otherwise I can close the PR.
I'm using this in my fork but might be we have other way to do that in otelfiber.
Some other go web frameworks have a similar feature for their OTel instrumentation.

**Comment by ReneWerner87** [comment]
Sorry i lost the focus on this, will check it in next days and then let you know that the decision is

**Comment by ReneWerner87** [comment]
first thank you for the effort
your feature works and offers added value

however, it would be better if we follow the lead of all other middlewares and add a config setting for a check method "Next: func(c *fiber.Ctx) bool"

We have this in all other middlewares and the users should be familiar with it.
https://github.com/search?q=repo%3Agofiber%2Ffiber%20Next%3A&type=code
https://docs.gofiber.io/api/middleware/csrf and so on

this would also give us more flexibility, because what if you don't want to use a url, but a parameter, cookie or header to skip this functionality.

i know in the 2 other contrib middlewares(Fiberzap, otelfiber) a skipurl config property was also added
this one i had unfortunately overlooked and i will add improvement reports/tasks to change this as deprecated and add a next method configuration property there as well

So that this concept is found in all middlewares and the consumers see a red trail

you would help us a lot if you could adjust your pull request accordingly

**Comment by emanuelef** [comment]
@ReneWerner87 Thanks for looking into that. Yes, with Next it would be much better, I'll try to adapt my PR to use that.
Not sure if it should be a separate PR and close this one.

File: otelfiber/config.go [hunk 1]
```diff
@@ -28,6 +29,12 @@ func (o optionFunc) apply(c *config) {
 	o(c)
 }
 
+func Next(f func(ctx *fiber.Ctx) bool) Option {
```
Comment (emanuelef) [inline otelfiber/config.go:32]: Not sure if to follow the pattern here should be WithNext, it's a bit different to the standard fibre middleware where we have ConfigDefault

Comment (ReneWerner87) [reply otelfiber/config.go:32]: name it withNext, its a shortcut function to work with the defaults

**Review by gaby — APPROVED** [review]
LGTM

**Comment by ReneWerner87** [comment]
@emanuelef 

- [ ] can you add the config possiblities to the README.md

https://github.com/gofiber/contrib/blob/main/otelfiber/README.md

like the others
https://github.com/gofiber/contrib/blob/main/opafiber/README.md#config
https://github.com/gofiber/contrib/blob/main/paseto/README.md#config
https://github.com/gofiber/contrib/tree/main/fibersentry#config
....
and maybe a example for the usage

we will use these readme's later in our https://docs.gofiber.io/ and there it is important to have all the information about this packet

this should show up as an issue report in the repository itself or discord channel

because only a well documented software can be well used

**Comment by emanuelef** [comment]
I've added two new sections to the README, but I have some questions there.
Most of the example seems to be inspired by another popular go web framework, I was wondering if we want to have something different. Also the tests are almost exactly the same and are not covering all of the possible configs (WithSpanNameFormatter is not tested).

**Comment by gaby** [comment]
@emanuelef Totally agree, that could be done in a separate PR so this one can be merged. :-)

**Pull request merged**